File: Tests/RegexTests.pas
namespace GoldLibrary.Tests.Shared;

uses
  RemObjects.Elements.EUnit,
  RemObjects.Elements.RTL;

type
  RegeexTests = public class(Test)

  public
    method SimpleMatchTest;
    begin
      Assert.IsTrue(TestApplication2.DoSimpleMatchTest());
    end;


    method MustCompileTest;
    begin
      Assert.IsTrue(TestApplication2.DoMustCompileTest());
    end;

  end;

end.

File: Tests/RegexTests.go
package GoldLibrary.Tests.Shared
import (
	"bytes"
	"fmt"
	"regexp"
)


func DoSimpleMatchTest() bool {
 matched, _ := regexp.MatchString("foo.*", "seafood")
 return matched
}

func DoMustCompileTest() bool {
	var validID = regexp.MustCompile(`^[a-z]+\[[0-9]+\]$`)
	return validID.MatchString("adam[23]")
}